onnx/onnx.proto

```diff
@@ -510,19 +510,49 @@ message TypeProto {
     optional TensorShapeProto shape = 2;
   }
 
+  // repeated T
+  message Sequence {
+    // The type and optional shape of each element of the sequence.
+    // This field MUST be present for this version of the IR.
+    optional TypeProto elem_type = 1;
```

Comment:
I am wondering if we still need this comment here? If this can be optional indeed, we can use Sequence to export heterogeneous tensors of different datatypes also by keeping `elem_type` unspecified.

Reply:
Supporting heterogeneous tensors can probably by adding another type `Tuple`. Sequence stills sounds like a homogeneous type like `IEnumerable<ElementType>` in C#.

Reply:
I believe Spandan is talking about an unbounded list/sequence, but where the elements don't have the same type. I think there are some details concerning such a type worth sorting out first. E.g., do we just want a list of tensors of different types, or do we want truly heterogeneous lists (a list where some elements are tensors, other elements are lists of tensors, other elements are maps, etc.)? I guess it should be the later to be truly general-purpose. More importantly, we would need support for some form of dynamic-cast from untyped values to make use of such a type. I think it would be helpful to have a more comprehensive proposal (e.g., take a simple example with heterogeneous tensors and see what's required to support it).

Reply:
In short, I think we can do that generalization separately, so that this PR can be kept simple.

Reply:
Yes, I think hetergeneous lists is a use case that we need to target. But I agree that it can designed separately to keep this PR simple.

Reply:
If we need heterogeneous types in one single variable, I'd personally consider a new type `Tuple` instead of `Sequence`. `Sequence` is more like a collection type where elements are not stored in consecutive memory blocks.